models/MARBL_joint_estimation/model_mod.f90

```diff
@@ -0,0 +1,654 @@
+
+! Flush the buffer and leave netCDF file open
+call nc_synchronize_file(ncid)
+
+end subroutine nc_write_model_atts
+
+!------------------------------------------------------------------
+! Verify that the namelist was filled in correctly, and check
+! that there are valid entries for the dart_kind.
+! Returns a table with columns:
+!
+! netcdf_variable_name ; dart_qty_string ; lowerbound ; upperbound ; update_string
+
+subroutine verify_state_variables(state_variables, ngood, table, qty_list, clamp_vals, update_var)
+
+character(len=*),  intent(inout) :: state_variables(:)
+integer,           intent(out) :: ngood
+character(len=*),  intent(out) :: table(:,:)
+integer,           intent(out) :: qty_list(:)   ! kind number
+real(r8),          intent(out) :: clamp_vals(:,:)
+logical,           intent(out) :: update_var(:) ! logical update
+
+integer :: nrows, i
+character(len=NF90_MAX_NAME) :: varname, dartstr, lowerbound, upperbound, update
+character(len=256) :: string1, string2
+
+if ( .not. module_initialized ) call static_init_model
+
+nrows = size(table,1)
+
+ngood = 0
+
+MyLoop : do i = 1, nrows
+
+    varname =    trim(state_variables(5*i -4))
+    dartstr =    trim(state_variables(5*i -3))
+    lowerbound = trim(state_variables(5*i -2))
+    upperbound = trim(state_variables(5*i -1))
+    update  =    trim(state_variables(5*i   ))
+
+    call to_upper(update)
+
+    table(i,1) = trim(varname)
+    table(i,2) = trim(dartstr)
+    table(i,3) = trim(lowerbound)
+    table(i,4) = trim(upperbound)
+    table(i,5) = trim(update)
+
+    if ( table(i,1) == ' ' .and. table(i,2) == ' ' .and. table(i,3) == ' ' .and. table(i,4) == ' ' .and. table(i,5) == ' ') exit MyLoop ! Found end of list.
+
+    if ( table(i,1) == ' ' .or. table(i,2) == ' ' .or. table(i,3) == ' ' .or. table(i,4) == ' ' .or. table(i,5) == ' ') then
+        string1 = 'model_nml:model_state_variables not fully specified'
+        call error_handler(E_ERR,'verify_state_variables',string1)
+    endif
+
+    ! Make sure DART qty is valid
+
+    qty_list(i) = get_index_for_quantity(dartstr)
+    if( qty_list(i)  < 0 ) then
+        write(string1,'(''there is no obs_kind <'',a,''> in obs_kind_mod.f90'')') trim(dartstr)
+        call error_handler(E_ERR,'verify_state_variables',string1)
+    endif
+    
+    ! Make sure the update variable has a valid name
+
+    select case (update)
+        case ('UPDATE')
+            update_var(i) = .true.
+        case ('NO_COPY_BACK')
+            update_var(i) = .false.
+        case default
+            write(string1,'(A)')  'only UPDATE or NO_COPY_BACK supported in model_state_variable namelist'
+            write(string2,'(6A)') 'you provided : ', trim(varname), ', ', trim(dartstr), ', ', trim(update)
+            call error_handler(E_ERR,'verify_state_variables',string1, text2=string2)
+    end select
+
+    ! reading the clamp values
+
+    if (table(i, 3) /= 'NA') then
+        read(table(i,3), '(d16.8)') clamp_vals(i,1)
```

Comment:
Is the format specification needed here? What if people added the clamps in a different format?